distributed-datawarehouse/duckdb/README.md

```diff
@@ -0,0 +1,253 @@
+    classDef storage fill:#fcf,stroke:#333,stroke-width:2px
+    
+    class Orchestrator primary
+    class USCompute1,USCompute2,USCompute3,EUCompute1,EUCompute2,EUCompute3 secondary
+    class USStorage,EUStorage,GlobalStorage storage
+```
+
+
+### Key Components
+
+- **Bacalhau Orchestrator**: Central control plane for job distribution
+- **Compute Nodes**: Distributed across regions, running close to data
+- **Local Storage**: Regional data stores (MinIO in this setup)
+- **DuckDB**: SQL query engine running on compute nodes
+
+## Why Bacalhau for Distributed Data Warehousing?
+
+Bacalhau addresses several key challenges in distributed data processing:
+
+1. **Compute-to-Data Movement**
+
+   - Deploy compute nodes close to data sources
+   - Minimize data movement across regions
+   - Process data where it lives
+
+2. **Distributed Orchestration**
+
+   - Central control plane for job management
+   - Region-aware job scheduling
+   - Automatic node selection and constraints
+
+3. **Efficient Partitioned Processing**
+
+   - Built-in data partitioning capabilities
+   - Parallel execution across nodes
+   - Automatic workload distribution
+
+4. **Query Engine Integration**
+   - Works with any query engine (DuckDB in this case)
+   - Extends single-instance engines to distributed scenarios
+   - Adds orchestration and parallel execution capabilities
+
+## DuckDB Integration
+
+We use DuckDB as our query engine for several reasons:
+
+- Excellent SQL support
+- Direct file reading capabilities
+- Fast analytical processing
+- Simple deployment (single binary)
+
+Bacalhau extends DuckDB's capabilities by:
+
+- Adding distributed orchestration
+- Enabling parallel processing
+- Managing partitioned execution
+- Handling job distribution
+
+### Custom DuckDB Image
+
+Our custom DuckDB image includes User-Defined Functions (UDFs) for partitioning:
+
+- `partition_by_hash`: Even distribution of files
+- `partition_by_regex`: Pattern-based partitioning
+- `partition_by_date`: Time-based partitioning
+
+## Data Access Patterns
+
+Bacalhau supports two ways to access local data:
+
+1. **Direct DuckDB Access**
+
+```sql
+SET VARIABLE files = (
+  SELECT LIST(file)
+  FROM partition_by_hash('s3://local-bucket/**/*.jsonl')
+);
+SELECT * FROM read_json_auto(getvariable('files'));
+```
+
+2. **Bacalhau Input Sources**
+
+```yaml
+InputSources:
+  - Type: s3
+    Source:
+      Bucket: local-bucket
+      Key: "data/*"
+```
+
+Each approach has its benefits:
+
+- Direct access is simpler for SQL-only jobs
+- Input sources provide more control and preprocessing options
+
+## Quick Start
+
+### Prerequisites
+- Bacalhau CLI
+- Docker and Docker Compose
+- Access to example multi-region setup
+
+### Deploy Environment
+```bash
+# Clone and start the multi-region environment
+git clone https://github.com/bacalhau-project/examples.git
+cd examples/setting-up-bacalhau-cluster/docker-compose/multi-region
+docker compose up -d
+```
+
```

Comment:
I know that we're going to have this here - but let's have a link to creating an account on Expanso Cloud

Reply:
actually I was thinking that maybe we can create another docker compose stack with just compute nodes, some local storage, and with an easy way to connect to expanso cloud? How does that sound?